internal/exec/stages/files/files.go

```diff
@@ -182,6 +177,15 @@ func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 				return err
 			}
 		}
+
+		if err := os.Chmod(path, os.FileMode(d.Mode)); err != nil {
+			return err
+		}
+
+		if err := os.Chown(path, *d.User.ID, *d.Group.ID); err != nil {
+			return err
+		}
+
```

Comment:
I don't particularly like the duplication of the logic a few lines up for this. I think a solution I would prefer would be to prepopulate `newPaths` with `path` and to then start [the loop](https://github.com/coreos/ignition/pull/422/files#diff-14bf0cdab0fa5f8140f06f308e1bca14R157) generating `newPaths` one dir up from `path`.

Something like this:

```
                newPaths := []string{path}
                for p := filepath.Dir(path); p != "/"; p = filepath.Dir(p) {

```